assets/global.js

```diff
@@ -533,10 +541,17 @@ class VariantSelects extends HTMLElement {
     const newMedia = document.querySelector(
       `[data-media-id="${this.dataset.section}-${this.currentVariant.featured_media.id}"]`
     );
+    const modalContent = document.querySelector('.product-media-modal__content');
+    const newMediaModal = modalContent.querySelector( `[data-media-id="${this.currentVariant.featured_media.id}"]`);
+
     if (!newMedia) return;
     const parent = newMedia.parentElement;
+    if (parent.firstChild == newMedia) return;
+    modalContent.prepend(newMediaModal);
     parent.prepend(newMedia);
-    window.setTimeout(() => { parent.scroll(0, 0) });
+    parent.dispatchEvent(this.mediaChangeEvent);
+    window.setTimeout(() => { parent.scrollLeft = 0; parent.scrollIntoView({behavior: "smooth", inline: "start"}); });
```

Comment:
The issue with it not scrolling vertically and horizontally does indeed appear to be a chrome bug, specifically with the 'smooth' behavior. A normal 'auto' behavior doesn't appear to have the issue, though it's not as slick as the smooth scrolling effect.

I did play with this a bit though and it seems like potentially targeting the first product media item (li) instead of the parent (ul) might allow both scrolls [to work](https://screenshot.click/09-56-mybot-cxnqn.mp4)

I used `document.querySelector('li.product__media-item').scrollIntoView({behavior: "smooth"});`, basically re-querying for the first item each time the media changes. Maybe you found an issue with this approach already, but just in case I wanted to mention that I had some success with it.